gibbon-compiler/src/Gibbon/Passes/CalculateBounds.hs

```diff
@@ -0,0 +1,158 @@
+          _      -> case ex of
+            LitE    _          -> err
+            FloatE  _          -> err
+            LitSymE _          -> err
+            ProjE{}            -> todo
+            TimeIt{}           -> todo
+            WithArenaE{}       -> todo
+            SpawnE{}           -> todo
+            SyncE{}            -> todo
+            MapE{}             -> todo
+            FoldE{}            -> todo
+            AppE _ _locs _args -> do
+              return (ex, regEnv)
+            PrimAppE{}             -> return (ex, regEnv)
+            DataConE loc dcon args -> do
+              (_, res) <- unzip <$> mapM go args
+              return (DataConE loc dcon args, mconcat res)
+            IfE cond bod1 bod2 -> do
+              (bod1', regEnv1) <- go bod1
+              (bod2', regEnv2) <- go bod2
+              return (IfE cond bod1' bod2', M.unionWith max regEnv1 regEnv2)
+            MkProdE ls -> do
+              (ls', regEnvs) <- unzip <$> mapM go ls
+              return (MkProdE ls', M.unionsWith max regEnvs)
+            LetE (v, locs, ty0, bind) bod -> do
+              (bind', regEnv') <- go bind
+              let venv' = M.insert v ty0 (vEnv env2)
+              (bod', regEnv'') <- calculateBoundsExp ddefs env2 { vEnv = venv' } szEnv locEnv regEnv bod
+              return (LetE (v, locs, ty0, bind') bod', M.unionWith max regEnv' regEnv'')
+            CaseE ex2 cases -> do
+              (cases', res) <-
+                unzip
+                  <$> mapM
+                        (\(dcon :: DataCon, vlocs :: [(Var, LocVar)], bod :: Exp2) -> do
+                          -- let offsets = M.fromList 
+                          --       . tail -- remove tag
+                          --       . scanl1 (\(_, s1) (v2, s2) -> (v2, s1+s2)) -- accumulate offsets
+                          --       . ((undefined, BoundedSize 1):)  -- add size for tag
+                          --       . zip (map snd vlocs) -- take locations
+                          --       . map (maybe Undefined BoundedSize . sizeOfTy) -- map to our region size type
+                          --       $ lookupDataCon ddefs dcon -- find ddef)
+                          -- * NOTE: After thinking, no location can be allocated from offset to an internal argument
+                          -- of a data constructor, except jumps so we can skip this analysis.
+                          (bod', re) <- go bod
+                          return ((dcon, vlocs, bod'), re)
+                        )
+                        cases
+              return (CaseE ex2 cases', M.unionsWith max res)
+            Ext ext -> case ext of
+              LetRegionE reg bod -> do
+                (bod', re) <- go bod
+                traceM $ ">> RegionSize: " ++ show reg ++ " -> " ++ show (re M.! regionToVar reg)
+                return (Ext $ LetRegionE (AnalyzedRegion reg (re M.! regionToVar reg)) bod', re)
+              LetParRegionE{}        -> todo
```

Comment:
Doesn't this have to be processed just like `LetRegionE`? Also, `todo` could be named `default`..

Reply:
Changing to pass, i.e. nothing will be done, since default is a reserved keyword.